code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -116,16 +117,6 @@
 		if(H.wear_suit != src)
 			return
 
-		if(attached_helmet && helmet)

```

Comment:
You should retain this in some form, otherwise people won't get a helmet when they first put the suit on, which is a desired function.